### PR TITLE
05 08 19 food delete

### PR DESCRIPTION
**Pull request by Mackenzie-Frey**
### Food Delete
This pull request implements functionality for a DELETE food request to an existing id in the database. Tests are not functional. I have spent a significant amount of time on the tests and it no longer makes sense to progress forward solo on the testing due to learning goals and time constraints. Intend to pair with someone tomorrow to get the tests up and running. The tests contain comments I would like to leave for now.

The endpoint is functional in Postman and I would like to merge into master. I made another User Story for testing the endpoint.

### User Story
#8 
As a user, when I send a `DELETE` request to `/api/v1/foods/:id`, it deletes the requested food item with the id provided in the params.

A successful response will return a 204 status code.

- [x] Successful response includes `204` status code.
- [x] Unsuccessful response returns a `404` status code. 

Endpoint is fully tested for:
- [ ] Successful response includes 204 status code.
- [ ] Food item is no longer in database.

- [ ] 404 status if id is not provided in request.
- [ ] 404 status if id does not match any existing record.
- [ ] Food item is still in database if not deleted due to unsuccessful request.

-------------------
- [x] Endpoint has been documented in the README
- [ ] Endpoint has been refactored

Functional in Postman. `DELETE` request:
image

Followed by a `GET` request for the same food `id`:
image

### CODE REVIEW

File: routes/api/v1/foods.js [hunk 1]
```diff
@@ -113,6 +114,22 @@ function validateRequest(req) {
   })
 }
 
+router.delete('/:id', async (req, res, next) => {
+  res.setHeader("Content-Type", "application/json");
+  try {
+    const food = await Food.findOne({where: {id: req.url.slice(1)}})
+    if (!food) {
```
Comment (nagerz) [inline routes/api/v1/foods.js:124]: I think we should return some kind of response for both of these outcomes. If a food is deleted, it should let the user know it is deleted. If a food is not found here, it should say the requested item could not be found.

File: routes/api/v1/foods.js [hunk 2]
```diff
@@ -113,6 +114,22 @@ function validateRequest(req) {
   })
 }
 
+router.delete('/:id', async (req, res, next) => {
+  res.setHeader("Content-Type", "application/json");
+  try {
+    const food = await Food.findOne({where: {id: req.url.slice(1)}})
```
Comment (nagerz) [inline routes/api/v1/foods.js]: Should be able to use 'req.params.id' instead of slicing the url.

File: specs/endpoints/food_delete.spec.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  describe('', () => {
+    test('it should delete a food successfully', () => {
+      // Below is an alternative way to create a food. Although, by utilizing this way,
+      // I am receiving the error that food cannot be created without a name, I don't understand
+      // why the given name of Cheese Puffs is not being utilized.
+
+      // const id = 83
+      // Food.create({
+      //   where: {
+      //     id: id,
```
Comment (nagerz) [inline specs/endpoints/food_delete.spec.js:27]: you don't need to supply an Id when creating a food

File: specs/endpoints/food_delete.spec.js [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+      return request(app).post("/api/v1/foods").send(newFood)
+      .then(response => {
+        expect(response.status).toBe(200)
+        const matchingFoods = Food.findAndCountAll({
+          where: {
+            name: "Pringles"
+          }
+        })
+          .then(foods => {
```
Comment (nagerz) [inline specs/endpoints/food_delete.spec.js:51]: Whether created manually or by post request, should be able to find the food by name and then grab the return food.id to store and send in your delete request.

File: specs/endpoints/food_delete.spec.js [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+      })
+
+
+
+
+    })
+
+// The below tests are passing, but are commented out for the sake of simplicity until the above if figured out.
+
```
Comment (nagerz) [inline specs/endpoints/food_delete.spec.js:80]: We can take a look at together

**Review by nagerz — COMMENTED** [review]
Think some small things could be added.

File: specs/endpoints/food_delete.spec.js (hunk 3 above)
Comment (Mackenzie-Frey) [reply specs/endpoints/food_delete.spec.js:27]: The intent of this was to create a food with an id that is consistent every time the test is run. I tried this with and without the id, and it still was not creating the food item with the name.

File: routes/api/v1/foods.js (hunk 2 above)
Comment (Mackenzie-Frey) [reply routes/api/v1/foods.js]: Good call. I went ahead and made the change.

File: routes/api/v1/foods.js (hunk 1 above)
Comment (Mackenzie-Frey) [reply routes/api/v1/foods.js:124]: I see what you're saying. For the `404` status I went ahead and implemented the following response ` {
        "error": "The requested food item could not be found and was therefore not deleted."
      }`. The `204` status code means `No Content` and the status code is what the project spec requests, therefore I think it's important to leave the `204` status code without content in the body.

File: specs/endpoints/food_delete.spec.js (hunk 4 above)
Comment (Mackenzie-Frey) [reply specs/endpoints/food_delete.spec.js:51]: I've tried this with several different approaches. Do you have functional code that does this?

File: routes/api/v1/foods.js (hunk 1 above)
Comment (nagerz) [reply routes/api/v1/foods.js:124]: I shortened it to to "The requested food item could not be found." That is more consistent with other endpoint responses.

Comment (nagerz) [reply routes/api/v1/foods.js:124]: I think no content in the body is a bad user experience but fine leaving it at least for now.

**Review by nagerz — APPROVED** [review]

**Comment by nagerz** [comment]
Reviewed and approved. Some items (successful responses and testing) to revisit at a later time.

**Pull request merged**